Makefile

```diff
@@ -76,6 +76,9 @@ run-e2e-tests:
 	# This is a test that should be run in the end to reduce the disruption to other tests because
 	# it will delete a node.
 	KUBECONFIG=${kubeconfig} go test -mod=$(MOD) -tags="$(platform),disruptive-e2e" -covermode=atomic -buildmode=exe -v ./...
+	# Test if the metrics are actually being scraped
+	KUBECONFIG=${kubeconfig} kubectl -n monitoring port-forward svc/prometheus-operator-prometheus 9090:9090 &
```

Comment:
Can't this forwarding be done from the Go code itself?

Reply:
There is a way to do it. But I wasn't sure if it was worth the effort. This felt easier and hence I implemented it. But going forward we can try to use that.

Reply:
Running that in the background while having Go available seems hacky to me, as it involves external dependencies into test suite.

Reply:
>  while having Go available

what do you mean here?

Reply:
You're writing and executing tests in Go, so I don't see a reason to also rely on some shell scripts running in the background. This make the test overly complex.

Reply:
done